recursive_retrieval/langchain_retrieval/retrieve.py

```diff
@@ -0,0 +1,66 @@
+import os
+from dotenv import load_dotenv
+from langchain_community.graphs import Neo4jGraph
+from langchain_community.vectorstores import Neo4jVector
+from langchain_openai import OpenAIEmbeddings
+from langchain_core.pydantic_v1 import BaseModel, Field
+from typing import List
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_openai import ChatOpenAI
+from langchain_community.vectorstores.neo4j_vector import remove_lucene_chars
+
+
+class Entities(BaseModel):
+    """Identifying information about entities."""
```

Comment:
Did you write all these docstring?
I love all the detailed descriptions!

Reply:
I wrote some of them, others are from a tutorial though